### ISSUE TITLE
Feature to define history window in config

### ISSUE THREAD

**Issue by nominandum** [mvisonneau/gitlab-ci-pipelines-exporter#38]
Awesome collector, thanks a million for providing this tool!

Would it be possible to define a date in the config, from where the scraper should start running? Currently every restart of the docker project starts off with only the latest commit present.

Again, thank you so much!

**Comment by mvisonneau** [comment]
Thanks @nominandum! I am not sure to understand your question though 🤔

It currently fetches the most recent pipelines, regardless of the fact of being restarted or not.

**Comment by nominandum** [comment]
Hi @mvisonneau sorry for not being clear and thanks for getting back on this!

It is currently not displaying any data from the past 
![Screenshot from 2019-12-05 12-51-06](https://user-images.githubusercontent.com/58513008/70233053-17177780-175e-11ea-9ecc-908b9300dac0.png)

we use gitflow, branches are removed post merge into master. there have been a lot of failed pipelines in the past, i promise ;) but they aren't showing up in the dashboard.

do we need to configure the exporter differently? currently its

`gitlab:
  url: https://gitlab.com
  token: -----

projects_polling_interval_seconds: 60
refs_polling_interval_seconds: 60
pipelines_polling_interval_seconds: 60
pipelines_max_polling_interval_seconds: 1800


projects:
  - name: --/--
  - name: --/--`

**Comment by mvisonneau** [comment]
Oh I see! I believe you might be able to overcome this issue by tweaking the dashboard though as those former metrics should still be available within your prometheus server. I'll put some thoughts on how to efficiently fetch all the existing pipelines from the api even if the refs do not exists anymore.

**Comment by mvisonneau** [comment]
I believe that the way to approach it would be to list the `refs` from the pipelines. Unfortunately the API does not allow to filter pipelines per branch or to search all refs from the pipelines.

If we would be adding the following args (and the associated logic of course), do you believe that would cover your use-case @nominandum?

```yaml
# Whether to attempt retrieving refs from pipelines when the exporter starts
on_init_fetch_refs_from_pipelines: true
# Maximum number of pipelines to analyze per project to search for refs
on_init_fetch_refs_from_pipelines_depth_limit: 100
```

**Comment by nominandum** [comment]
yes, that would perfectly solve our problem. i'm sure we'll find a way to filter the refs somewhere after the collector. but even without, this would significantly enhance the overview we're getting from the dashboard :)

**Comment by mvisonneau** [comment]
I have implemented and released the feature in `0.2.10`

Let me know if you have issues!
